src/urllib3/connectionpool.py

```diff
@@ -633,8 +654,8 @@ def urlopen(
         # have to copy the headers dict so we can safely change it without those
         # changes being reflected in anyone else's copy.
         if not http_tunnel_required:
-            headers = headers.copy()
-            headers.update(self.proxy_headers)
+            headers = headers.copy()  # type: ignore
```

Comment:
Ugh, this type: ignore is annoying because our APIs are clearly favoring `dict` instead of `Mapping`. Can't wait to switch to `HTTPHeaderDict` everywhere...